internal/replacements/replacements.go

```diff
@@ -188,22 +188,27 @@ func processGroupNeedsRemoval(cluster *fdbv1beta2.FoundationDBCluster, pod *core
 		}
 	}
 
-	expectedNodeSelector := cluster.GetProcessSettings(processClass).PodTemplate.Spec.NodeSelector
-	if !equality.Semantic.DeepEqual(pod.Spec.NodeSelector, expectedNodeSelector) {
-		specHash, err := internal.GetPodSpecHash(cluster, processClass, idNum, nil)
-		if err != nil {
-			return false, err
-		}
+	// When three data hall redundancy is enabled. The node selecctor is not provided by the spec.
```

Comment:
What if a user wants to define a nodeSelector and wants to use three data hall? e.g. a user wants to ensure that Pods are running on a node labels `foo=bar`.